src/chrome/extension/scripts/chrome_browser_api.ts

```diff
@@ -201,10 +201,16 @@ class ChromeBrowserApi implements BrowserAPI {
     this.popupState_ = PopupState.LAUNCHED;
   }
 
-  public showNotification = (notificationText :string) => {
+  public showNotification = (text :string, tag :string) => {
     var notification =
-        new Notification('uProxy', { body: notificationText,
-                         icon: 'icons/38_' + UI.DEFAULT_ICON});
+        new Notification('uProxy', {
+          body: text,
+          icon: 'icons/38_' + UI.DEFAULT_ICON,
+          tag: tag
+        });
+    notification.onclick = function() {
+      ui.handleNotificationClick(this.tag);
```

Comment:
Copying offline discussion:

I don't really see the point in blocking this entire patch behind an issue with a couple-line merge conflict.  If it becomes too big of an issue, I can handle the merge for you on the other branch.